File: CS/PS/Site/leetcode/easy/21.py
class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
      newNode = ListNode(0)
      if(not (l1 and l2)): return newNode
      while(l1 != None and l2 != None):
        if(l1.val > l2.val):
          newNode.next = l2
          l2 = l2.next
        else:
          newNode.next = l1
          l1 = l1.next
      while(l1):
        newNode.next = l1
        l1 = l1.next
      while(l2):
        newNode.next = l2
        l1 = l2.next
      return newNode

File: CS/PS/Site/leetcode/easy/21.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var mergeTwoLists = function (l1, l2) {
  let head = { val: 0, next: null };
  let curr = head;

  while (l1 && l2) {
    if (l1.val > l2.val) {
      curr.next = l2;
      l2 = l2.next;
    } else {
      curr.next = l1;
      l1 = l1.next;
    }
    curr = curr.next;
  }
  curr.next = l1 || l2;
  return head.next;
};
